airbyte-api/src/main/openapi/config.yaml

```diff
@@ -4734,6 +4734,9 @@ components:
           additionalProperties: true # Oauth parameters like code, state, etc.. will be different per API so we don't specify them in advance
         oAuthInputConfiguration:
           $ref: "#/components/schemas/OAuthInputConfiguration"
+        returnSecretCoordinate:
```

Comment:
would recommend setting the default to false here to avoid impacting existing flows

Reply:
we will probably also need to update the return type signature

Reply:
Good point, default updated.

I believe updating the return type signature isn't necessary because we return a Map<String, Object> and we return the same when returning a secret ID. The API spec reflect this (since the return of OAuth can be varied, it's just a generic Map).

W/ regards to the first conditional, nobody is using coordinates instead of refresh tokens currently during source creation (as it's not a supported flow) and so there should be no consumers who would expect to be able to do this. If someone used the new flag to get a secret ID through completing OAuth and then attempted to post it to connections in an unsupported way they should fail validation, but there's nothing that would prevent them from giving us a secret ID instead of access/refresh tokens if they really wanted to. However, that generally amounts to them just passing us an invalid access token/refresh token and they shouldn't expect the source to get set up correctly anyway. In current state there should be no nesting of secrets incorrectly.

This  endpoint is just a building block of a new flow in which we complete OAuth, receive a secret ID, and later use that secret ID during source creation instead of passing the raw tokens. The usage of a secret ID during source creation instead of credentials of any kind is not yet developed, but will be shortly.